### PR TITLE
Add HPOS support with backward compatibility

### PR DESCRIPTION
**Pull request by mralaminahamed**
### Closes 
* Closes https://github.com/getdokan/woocommerce-conversion-tracking/issues/45

## Summary by CodeRabbit

- **New Features**
	- Enhanced compatibility declarations for WooCommerce features, including HPOS support.
	- Introduced a new function to check if HPOS is enabled.
	- Updated integration manager to include additional integrations and clarified return types for methods.

- **Bug Fixes**
	- Refined order processing logic to ensure compatibility with various WooCommerce versions.

- **Documentation**
	- Updated method return types for better clarity and type safety.

- **Refactor**
	- Streamlined conversion tracking code handling and improved method structures.

### CODE REVIEW

File: includes/integration.php [hunk 1]
```diff
@@ -109,10 +109,18 @@ function validate_textarea_field( $key, $value ) {
      * @param int $post_id
      */
     function product_options_save( $post_id ) {
-
         if ( isset( $_POST['_wc_conv_track'] ) ) {
             $value = trim( sanitize_text_field( wp_unslash( $_POST['_wc_conv_track'] ) ) );
-            update_post_meta( $post_id, '_wc_conv_track', $value );
+
+            if ( wcct_is_hpos_enabled() ) {
```
Comment (mrabbani) [inline includes/integration.php:115]: We should not check `hpos` to update the product meta. We can remove the conditions.
```php
  $product->update_meta_data( '_wc_conv_track', $value );
   $product->save();
```

File: includes/integration.php [hunk 2]
```diff
@@ -225,25 +233,54 @@ function code_handler() {
     public function thankyou_page( $order_id ) {
         $order = wc_get_order( $order_id );
 
-        if ( $items = $order->get_items() ) {
-            foreach ( $items as $item ) {
-                $product = $order->get_product_from_item( $item );
+        error_log( print_r( $order, true ) );
```
Comment (mrabbani) [inline includes/integration.php:236]: Pls remove the `error_log`

File: includes/integration.php [hunk 3]
```diff
@@ -225,25 +233,54 @@ function code_handler() {
...
+
+        if ( ! $product ) {
+            return '';
+        }
+
+        if ( wcct_is_hpos_enabled() ) {
+            return $product->get_meta( '_wc_conv_track' );
+        } else {
+            return get_post_meta( $product_id, '_wc_conv_track', true );
+        }
```
Comment (mrabbani) [inline includes/integration.php:280]: Pls remove the `hpos` checking. Just access meta through `$product` object.

```php
    return $product->get_meta( '_wc_conv_track' );
```

File: includes/integration.php [hunk 4]
```diff
@@ -303,9 +343,15 @@ function process_order_markdown( $code ) {
...
+            $used_coupons = $order->get_used_coupons();
+
+        } else {
+            $used_coupons = $order->get_coupon_codes();
+        }
+
         $customer       = $order->get_user();
-        $used_coupons   = $order->get_used_coupons() ? implode( ',', $order->get_used_coupons() ) : '';
-        $order_currency = $order_currency;
+        $used_coupons   = implode( ',', $order->$used_coupons() );
```
Comment (mrabbani) [inline includes/integration.php:354]: Pls remove this statement

File: includes/class-abstract-integration.php [hunk 5]
```diff
@@ -80,7 +80,7 @@ public function get_name() {
     public function is_enabled() {
         $settings = $this->get_integration_settings();
 
-        if ( $settings && $settings[ 'enabled' ] == true ) {
+        if ( $settings && $settings[ 'enabled' ] === true ) {
```
Comment (mrabbani) [inline includes/class-abstract-integration.php:83]: Type cast `$settings[ 'enabled' ]` to boolean.

**Review by mrabbani — CHANGES_REQUESTED** [review]

File: includes/integration.php (hunk 2 above)
Comment (mralaminahamed) [reply includes/integration.php]: vai, done

File: includes/integration.php (hunk 1 above)
Comment (mralaminahamed) [reply includes/integration.php]: done

File: includes/integration.php (hunk 4 above)
Comment (mralaminahamed) [reply includes/integration.php]: vai, done

**Review by mrabbani — APPROVED** [review]

**Pull request merged**